### PR TITLE
add filename filter to modify path and filename

### PR DESCRIPTION
**Pull request by lcxfs1991**
sometimes you need to make the entry like the following in order to put bundle and vendor in different folders. 

```
entry: {
        "./js/index": [path.join(config.path.src, "/page/index/main.js")],
        "./js/spa": [path.join(config.path.src, "/page/spa/main.js")],
        "./libs/react": [path.join(config.path.src, "/libs/react.js")],
},
```

However, this will lead to strange css destination folder.
csspath

Adding this filenamefilter function can help to solve this problem manually by developers themselves.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,6 +64,15 @@ new ExtractTextPlugin(options: filename | object)
 * `options.allChunks: boolean` extract from all additional chunks too (by default it extracts only from the initial chunk(s))
 * `options.disable: boolean` disables the plugin
 * `options.id: string` Unique ident for this plugin instance. (For advanced usage only, by default automatically generated)
+* `filenamefilter` function to modify path and filename before files are emitted
+``` javascript
+{
+  filenamefilter: function(filename) {
```
Comment (michael-ciniawsky) [inline README.md]: `filenamefilter` => `filter` or why not just letting `options.filename` accepts a function ?

File: package.json [hunk 2]
```diff
@@ -24,10 +24,10 @@
     "style-loader": "^0.13.0",
     "webpack": "^2.1.0-beta"
   },
-  "homepage": "http://github.com/webpack/extract-text-webpack-plugin",
+  "homepage": "http://github.com/lcxfs1991/extract-text-webpack-plugin",
```
Comment (michael-ciniawsky) [inline package.json]: Should stay `webpack` unless your are not planning to fork and release

File: package.json [hunk 3]
```diff
@@ -24,10 +24,10 @@
...
   "repository": {
     "type": "git",
-    "url": "http://github.com/webpack/extract-text-webpack-plugin.git"
+    "url": "http://github.com/lcxfs1991/extract-text-webpack-plugin.git"
```
Comment (michael-ciniawsky) [inline package.json]: Same here 😛

File: package.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "extract-text-webpack-plugin",
+  "name": "extract-text-webpack-plugin-steamer",
```
Comment (michael-ciniawsky) [inline package.json]: Should stay `extract-text-webpack-plugin` in PR unless your not to planning to release your fork

**Comment by michael-ciniawsky** [comment]
@lcxfs1991 I this PR superseded by #269 ?

@bebraw Coud you add #203 / #269  to the v2 milestone please?

**Comment by bebraw** [comment]
Let's close this in favor of #269 and continue there.

**Pull request closed without merging**